packages/backend/server/src/plugins/oauth/providers/oidc.ts

```diff
@@ -0,0 +1,203 @@
+        method: 'GET',
+        headers: {
+          Accept: 'application/json',
+        },
+      });
+
+      if (!response.ok) {
+        throw new Error(
+          `Failed to fetch OIDC configuration: ${response.statusText}`
+        );
+      }
+
+      const fullConfig = await response.json();
+      this.oidcConfig = {
+        authorization_endpoint: fullConfig.authorization_endpoint,
+        token_endpoint: fullConfig.token_endpoint,
+        userinfo_endpoint: fullConfig.userinfo_endpoint,
+        end_session_endpoint: fullConfig.end_session_endpoint,
+      };
+    } catch (error) {
+      this.logger.error(
+        `Failed to fetch OIDC configuration: ${(error as Error).message}`
+      );
+    }
+  }
+
+  private mapUserInfo(
+    user: Record<string, any>,
+    claimsMap: Record<string, string>
+  ): UserInfo {
+    const mappedUser: Partial<UserInfo> = {};
+    for (const [key, value] of Object.entries(claimsMap)) {
+      if (user[value] !== undefined) {
+        mappedUser[key as keyof UserInfo] = user[value];
+      }
+    }
+    return mappedUser as UserInfo;
+  }
+
+  private checkOIDCConfig(
+    oidcConfig: OIDCConfiguration | null
+  ): asserts oidcConfig is OIDCConfiguration {
+    if (!oidcConfig) {
+      throw new Error('OIDC configuration has not been loaded yet.');
+    }
+  }
+
+  getAuthUrl(state: string): string {
+    this.checkOIDCConfig(this.oidcConfig);
+    return `${this.oidcConfig.authorization_endpoint}?${this.url.stringify({
+      client_id: this.config.clientId,
+      redirect_uri: this.url.link('/oauth/callback'),
+      response_type: 'code',
+      ...this.config.args,
+      state,
+    })}`;
+  }
+
+  async getToken(code: string): Promise<{
+    accessToken: string;
+    refreshToken: string;
+    expiresAt: Date;
+    scope: string;
+  }> {
+    this.checkOIDCConfig(this.oidcConfig);
+    try {
+      const response = await fetch(this.oidcConfig.token_endpoint, {
+        method: 'POST',
+        body: this.url.stringify({
+          code,
+          client_id: this.config.clientId,
+          client_secret: this.config.clientSecret,
+          redirect_uri: this.url.link('/oauth/callback'),
+          grant_type: 'authorization_code',
+        }),
+        headers: {
+          Accept: 'application/json',
+          'Content-Type': 'application/x-www-form-urlencoded',
+        },
+      });
+
+      if (response.ok) {
+        const tokenResponse = (await response.json()) as OIDCTokenResponse;
+
+        return {
+          accessToken: tokenResponse.access_token,
+          refreshToken: tokenResponse.refresh_token,
+          expiresAt: new Date(Date.now() + tokenResponse.expires_in * 1000),
+          scope: tokenResponse.scope,
+        };
+      } else {
+        const errorResponse = await response.json();
+        throw new Error(
+          `Server responded with non-success code ${response.status}, ${JSON.stringify(errorResponse)}`
```

Comment:
too detailed for returned error, i prefer print the full error in console, and return the error category and status to frontend